### ISSUE TITLE
New installation on xubuntu 24.04, just get "404 Not Found" when I try to login

### ISSUE THREAD

**Issue by chrisisbd** [gerbera/gerbera#3243]
I'm a new user of gerbera.  I have installed version 2.3.0 from the Gerbera repository and have started it using systemctl.

I get the welcome screen at 192.168.1.5:49152 but when I click login I just get a "404 Not Found".

Running 'gerbera --check-config' shows no errors:-
```

root@q957# gerbera --check-config
2024-10-08 12:16:19   info: Gerbera UPnP Server 2.3.0 - https://gerbera.io/
2024-10-08 12:16:19   info: Copyright 2016-2024 Gerbera Contributors.
2024-10-08 12:16:19   info: Gerbera is free software, covered by the GNU General Public License version 2
2024-10-08 12:16:19   info: Home detected as: /root
2024-10-08 12:16:19   info: Loading configuration from: /root/.config/gerbera/config.xml
2024-10-08 12:16:19   info: Parsing configuration...
2024-10-08 12:16:19   info: Using home directory '/root'
2024-10-08 12:16:19   info: Configuration load succeeded.
2024-10-08 12:16:19   info: Validating configuration...
2024-10-08 12:16:19   info: Configuration check succeeded.

```

One oddity is that that ps shows:-

```
ps -ef | grep gerbera
gerbera   103765       1  1 12:15 ?        00:00:00 /usr/bin/gerbera -c /etc/gerbera/config.xml --syslog
```
But in reality the config file seems to be /root/.config/gerbera/config.xml.

Presumably I have something basic wrong somewhere but I can't see it at the moment.

**Comment by chrisisbd** [comment]
Oops, sorry to answer my own question, but it does seem to be a bug.

I saw the following in syslog:-
```
error: Exception: [/build/src/util/grb_fs.cc:166] FILE* GrbFile::open(const char*, bool) Error: Could not open /usr/share/gerbera/web/vendor/jquery/jquery.min.js
error: Exception: [/build/src/util/grb_fs.cc:166] FILE* GrbFile::open(const char*, bool) Error: Could not open /usr/share/gerbera/web/vendor/tether/tether.min.js
```
On looking in the share directory there were specific versions of those files but not the 'general' on.  I've created symbolic links for jquery.min.js and tether.min.js.  The WEB GUI now starts OK.

I'm still seeing the following error in syslog:-
```
Error: Could not open /usr/share/gerbera/web/vendor/font-awesome/fonts/fontawesome-webfont.wof
f2?v=4.7.0: No such file or directory
```
However this doesn't seem to break anything.

**Comment by KarlStraussberger** [comment]
You seem to have two different configuration files. One under `/root/.config/gerbera/config.xml` and the other under `/etc/gerbera/config.xml`.

Also the entry for `<webroot>` seems to point to the wrong path so that the `index.html` is not of the correct version, because we are pointing to the following files
```
    <script src="vendor/jquery/jquery-3.7.1.min.js" type="text/javascript"></script>
    <script src="vendor/js-cookie.min.js" type="text/javascript"></script>
    <script src="vendor/jquery/jquery-ui.min.js" type="text/javascript"></script>
    <script src="vendor/tether/tether-2.0.0.min.js" type="text/javascript"></script>
```

**Comment by chrisisbd** [comment]
Yes, OK.  Clearing things up and clearing the browser cache has got me to a working GUI now.

However, the Gerbera version in the Ubuntu 24.04 repositories (2.0.0) doesn't install correctly.  I installed that on a clean system and got the '404 Not Found' problem.  I got the above problem because installing 2.3.0 (having purged 2.0.0) suffered from bits of 2.0.0 left over (in the file system and in the browser cache).

I have anyway decided now that Gerbera is overkill for my requirements.

Thanks for your help anyway.

**Comment by ed-tipiglen** [comment]
I've used gerbera since mediatomb days with very happy results, but have just changed to a new computer-as-server, and now having trouble getting it to work.

I installed from the ubuntu 24.04 repository and am also getting a 404 on attempted login.
 Running on a remote computer on my home network.
 (Release Linux Mint 22 Wilma 64-bit - Kernel Linux 6.8.0-47-generic x86_64 - MATE 1.26.2)

I have tried running gerbera from terminal, both as su and ordinary user.   same results.  TV shows empty files for video, picture, and music.
Running at the moment (started through "software manager" because for some reason it doesn't appear in mint menu)
Still getting 404 errors on trying to log in, etc., and no sign of the filesystem/database, just the top banner
![image](https://github.com/user-attachments/assets/7f9e0759-5abb-4a62-b4f0-97e6e5847b35)


ed@ed-OptiPlex-7020:~$ gerbera
2024-10-22 10:41:06   info: Gerbera UPnP Server 2.0.0 - http://gerbera.io/
2024-10-22 10:41:06   info: Copyright 2016-2023 Gerbera Contributors.
2024-10-22 10:41:06   info: Gerbera is free software, covered by the GNU General Public License version 2
2024-10-22 10:41:06   info: Home detected as: /home/ed
2024-10-22 10:41:06   info: Loading configuration from: /home/ed/.config/gerbera/config.xml
2024-10-22 10:41:06   info: Parsing configuration...
2024-10-22 10:41:06   info: Using home directory '/home/ed'
2024-10-22 10:41:06   info: Configuration load succeeded.
2024-10-22 10:41:06   info: Validating configuration...
2024-10-22 10:41:06   info: Configuration check succeeded.
2024-10-22 10:41:06   info: Loading 0 configuration items from database
2024-10-22 10:41:06   info: Initialising UPnP with interface: <unset>, port: <unset>
2024-10-22 10:41:06   info: IPv4: Server bound to: 192.168.1.188:49153
2024-10-22 10:41:06   info: IPv6: Server bound to: fe80::cf66:b5aa:32b5:33df:49153
2024-10-22 10:41:06   info: IPv6 ULA/GLA: Server bound to: fd65:76f8:30e8:1:e6b3:c71:2414:8624:49153
2024-10-22 10:41:06   info: The Web UI can be reached by following this link: http://192.168.1.188:49153
2024-10-22 10:41:27  error: AutoscanInotify::monitorUnmonitorRecursive /home/ed/Video-Archive/: Failed to read /home/ed/Video-Archive/00ASecondAnnex, No such file or directory
2024-10-22 10:41:28  error: AutoscanInotify::monitorUnmonitorRecursive /home/ed/Video-Archive/: Failed to read /home/ed/Video-Archive/00ALink to annex, No such file or directory
2024-10-22 10:41:36  error: AutoscanInotify::monitorUnmonitorRecursive /home/ed/Video-Archive/: Failed to read /home/ed/Video-Archive/00A  link to SecondAnnex, No such file or directory
2024-10-22 10:43:00 warning: Caught Exiv2 exception processing /home/ed/Pictures-Archive/pictures/photos-not-core/Taliesin-simon-brooke/2004/10.30/shot.xcf: '/home/ed/Pictures-Archive/pictures/photos-not-core/Taliesin-simon-brooke/2004/10.30/shot.xcf: The file contains data of an unknown image type'
2024-10-22 10:51:24  error: iconv: Invalid or incomplete multibyte or wide character
2024-10-22 10:51:24  error: iconv: File written by Adobe Photoshop\ufffd 4.0 could not be converted to new encoding: invalid character sequence!
2024-10-22 10:51:24  error: iconv: Invalid or incomplete multibyte or wide character
2024-10-22 10:51:24  error: iconv: File written by Adobe Photoshop\ufffd 4.0 could not be converted to new encoding: invalid character sequence!
2024-10-22 10:51:24  error: iconv: Invalid or incomplete multibyte or wide character
2024-10-22 10:51:24  error: iconv: File written by Adobe Photoshop\ufffd 4.0 could not be converted to new encoding: invalid character sequence!
2024-10-22 10:51:24  error: iconv: Invalid or incomplete multibyte or wide character
2024-10-22 10:51:24  error: iconv: File written by Adobe Photoshop\ufffd 4.0 could not be converted to new encoding: invalid character sequence!
2024-10-22 10:51:24  error: iconv: Invalid or incomplete multibyte or wide character
2024-10-22 10:51:24  error: iconv: File written by Adobe Photoshop\ufffd 4.0 could not be converted to new encoding: invalid character sequence!
2024-10-22 10:51:25  error: iconv: Invalid or incomplete multibyte or wide character
2024-10-22 10:51:25  error: iconv: File written by Adobe Photoshop\ufffd 4.0 could not be converted to new encoding: invalid character sequence!
2024-10-22 10:51:25  error: iconv: Invalid or incomplete multibyte or wide character
2024-10-22 10:51:25  error: iconv: File written by Adobe Photoshop\ufffd 4.0 could not be converted to new encoding: invalid character sequence!
2024-10-22 10:51:26  error: iconv: Invalid or incomplete multibyte or wide character
......
